### ISSUE TITLE
Overview of how to interact with Astro Deployments Programmatically - Auth Tokens

### ISSUE THREAD

**Issue by manmeetkaur** [astronomer/docs#2305]
We currently have 3 ways of authenticating to Astro using CLI:

- Organization API Token
- Workspace API Token
- Deployment API Keys

Each can serve a different purpose and the wider scope token can also be utilized for the lesser scope object in Astro with appropriate permissions. Currently, what is missing from our docs is an overview of how to Authenticate and what you can use when, and the commonly used permutations based on Persona.

Also, I have observed that we are repeating a lot many times in our docs how to "automate access to deployment". For example, 
```
Set the following in your <tool>'s environment

ASTRONOMER_KEY_ID = <your-key-id>
ASTRONOMER_KEY_SECRET = <your-key-secret>
```

Now that we have multiple way of authenticating, we can put that info in one place and refer it in all the docs that need info.

**Comment by jwitz** [comment]
This converges with another issue, which is that many customers are asking for examples of CLI scripting. I think we should create a new doc similar to https://cloud.google.com/sdk/docs/scripting-gcloud that:

- Explains how to authorize the CLI
- Gives general best practices for scripting
- Includes examples of common scripting actions

**Comment by manmeetkaur** [comment]
Resolved as part of #2595 . Will create a follow-up issue with examples for CI/CD
